### ISSUE TITLE
(aws-ecs): Provide an enableExecutionCommand method

### ISSUE THREAD

**Issue by rcollette** [aws/aws-cdk#15768]
When the `enableExecutionCommand` property was added, the `enableExecutionCommand()` method was made private.  The logic that handles this property should have also been exposed as a method so that this feature could be enabled subsequent to construction of the construct.

### Use Case

For use cases where we do not have direct access to construction of the construct, such as in the case of `ApplicationLoadBalancedFargateService`.

### Proposed Solution

Move all logic related handling of the `enableExecutionCommand` configuration property into the `enableExecutionCommand()` method and make that method public.

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by ryparker** [comment]
Thanks for creating this and separating it from the other linked issue #15769. 👍🏻 

I'm curious to see what @madeline-k says, but for now I will prioritize this as p1 since you described in the issue that this was previously working as a public method and was changed to private in a recent update.

If anyone is visiting this issue and looking for a quick workaround please take a look at the docs on using [escape hatches](https://docs.aws.amazon.com/cdk/latest/guide/cfn_layer.html#Modifying-the-AWS-CloudFormation-resource-behind-AWS-constructs).

Example of using escape hatches on the `ApplicationLoadBalancedFargateService`:

```ts
const albFargateService = new ApplicationLoadBalancedFargateService(this, 'AlbFargateService', {
	...props
});

// Get the CloudFormation resource
const cfnFargateService = albFargateService.service.node;
cfnFargateService.addPropertyOverride('EnableExecuteCommand', true);
```

**Comment by rcollette** [comment]
@ryparker I apologize if I wasn't clear.   The method I mentioned was never public.  It was "made" in the first place private.   So probably not as much a bug (not a breaking one at least)

**Comment by ryparker** [comment]
> @ryparker I apologize if I wasn't clear. The method I mentioned was never public. It was "made" in the first place private. So probably not as much a bug (not a breaking one at least)

Gotcha thanks for clarifying.

**Comment by madeline-k** [comment]
@ryparker Agree with your triage here. And thanks for providing a workaround!

**Comment by LukvonStrom** [comment]
To quote from https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-exec.html: "You can't enable ECS Exec for existing tasks. It can only be enabled for new tasks.", so I don't see added benefit with this method

**Comment by hoegertn** [comment]
Yes, but calling a method on a construct still happens during synth before creating infrastructure. From a CFN perspective, there is no difference between settings props in a constructor or a method call.

**Comment by LukvonStrom** [comment]
@hoegertn are there any L2 constructs, that expose methods for props, that may be set during creation?

**Comment by hoegertn** [comment]
Depends on what you count as props. `grantXX`, `addToXXPolicy`, `allowFrom/To` and more I could look up.

**Comment by LukvonStrom** [comment]
@hoegertn these are side-effect causing methods, which have a legitimate purpose (imho), but I don't see that for this method here

**Comment by hoegertn** [comment]
Would love if this could happen
